internal/provider/resources/concurrency_limit.go

```diff
@@ -0,0 +1,263 @@
+type ConcurrencyLimitResourceModel struct {
+	BaseModel
+
+	AccountID   customtypes.UUIDValue `tfsdk:"account_id"`
+	WorkspaceID customtypes.UUIDValue `tfsdk:"workspace_id"`
+
+	Tag              types.String `tfsdk:"tag"`
+	ConcurrencyLimit types.Int64  `tfsdk:"concurrency_limit"`
+}
+
+// NewConcurrencyLimitResource returns a new ConcurrencyLimitResource.
+//
+//nolint:ireturn // required by Terraform API
+func NewConcurrencyLimitResource() resource.Resource {
+	return &ConcurrencyLimitResource{}
+}
+
+// Metadata returns the resource type name.
+func (r *ConcurrencyLimitResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_concurrency_limit"
+}
+
+// Configure initializes runtime state for the resource.
+func (r *ConcurrencyLimitResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(api.PrefectClient)
+	if !ok {
+		resp.Diagnostics.Append(helpers.ConfigureTypeErrorDiagnostic("resource", req.ProviderData))
+
+		return
+	}
+
+	r.client = client
+}
+
+// Schema defines the schema for the resource.
+func (r *ConcurrencyLimitResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "The resource `concurrency_limit` represents a concurrency limit for a tag.",
+		Version:     0,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Concurrency limit ID (UUID)",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"created": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Timestamp of when the resource was created (RFC3339)",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"updated": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Timestamp of when the resource was updated (RFC3339)",
+			},
+			"account_id": schema.StringAttribute{
+				Optional:    true,
+				Description: "Account ID (UUID)",
+				CustomType:  customtypes.UUIDType{},
+			},
+			"workspace_id": schema.StringAttribute{
+				Optional:    true,
+				Description: "Workspace ID (UUID)",
+				CustomType:  customtypes.UUIDType{},
+			},
+			"tag": schema.StringAttribute{
+				Required:    true,
+				Description: "Tag",
+				PlanModifiers: []planmodifier.String{
+					// Concurrency limit updates are not supported so any changes to the tag will
+					// require a replacement of the resource.
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"concurrency_limit": schema.Int64Attribute{
+				Required:    true,
+				Description: "Concurrency limit",
```

Comment:
```suggestion
				Description: "The concurrency limit.",
```

Small but may as well match the API docs field while we're here